source/val/validate_arithmetics.cpp

```diff
@@ -187,7 +226,8 @@ spv_result_t ArithmeticsPass(ValidationState_t& _, const Instruction* inst) {
 
     case spv::Op::OpMatrixTimesScalar: {
       if (!_.IsFloatMatrixType(result_type) &&
-          !_.IsCooperativeMatrixType(result_type))
+          !(_.IsCooperativeMatrixNVType(result_type) ||
+            _.IsCooperativeMatrixKHRType(result_type)))
```

Comment:
Isn't the original sufficient?

Reply:
IsFloatMatrixType this checks whether it is matrix (mat4 for example) and last two check whether it is Cooperative Matrix these (like coopmat<int8_t, gl_ScopeSubgroup, M, K, gl_MatrixUseA> matA;). Thus answer is: no, not sufficient.

Reply:
The original I was referring to was `IsCooperativeMatrix`.